src/v/kafka/protocol/response_writer.h

```diff
@@ -340,8 +340,7 @@ class response_writer {
         for (auto& [id, tag] : tags) {
             // write tag id +  size in bytes + tag itself
             write_unsigned_varint(id);
-            write_unsigned_varint(tag.size_bytes());
-            write_direct(std::move(tag));
+            write_size_prepended(std::move(tag));
```

Comment:
hmm, confusingly i think it could be replaced by `write_array` because the array is prefixed with a non varlen integer to denote number of tags.
```
    // Only relevent when writing flex responses
    uint32_t write_tags(tagged_fields&& tags) {
        return write_array(
          tags, [](std::tuple<uint32_t, iobuf>& p, response_writer& rw) {
              rw.write_unsigned_varint(std::get<0>(p));
              rw.write_size_prepended(std::move(std::get<1>(p)));
          });
    }
```
tests pass with this, but would it introduce confusion?

Reply:
Hmm, main difference seems to be:
```

Reply:
I'm thinking something like:
```c++
    uint32_t write_tags(tagged_fields&& tags) {
        return write_flex_array(tags, [](auto&& t){
            // write tag id +  size in bytes + tag itself
            write_unsigned_varint(id);
            write_size_prepended(std::move(tag));
          });
```

Reply:
But it looks like `write_flex_array` has `v.size() + 1`, because reasons?

Is it worth testing that the varint is correct by sending more than 127 items?

Reply:
Yeah weird, i would assume `write_nullable_flex_array` to have this `size + 1` and the non-nullable version to not, but i guess to avoid confusion they kept it consistent.

That's a good idea, i can encode larger strings then in the test code.